File: 2010/20101009_devinrio2_jokenpo/jokenpo.rb
def jokenpo(jogada1, jogada2)

    if jogada1 != jogada2
        if jogada1 == 'pedra' and jogada2 == 'tesoura'
            return 'pedra'
        end
        if jogada1 == 'tesoura' and jogada2 == 'tesoura'
            return 'pedra'
        end
        return 'pedra'
    end
    return 'empate'

end

File: aleatorios/javascript_jokenpo/jokenpo.js
function jokenpo (jogada1, jogada2) {
    var tabelaDeJogadas = {
    "pedra":{
        "pedra":"empate",
        "papel":"papel",
        "tesoura":"pedra"
    },
    "papel":{
        "pedra":"papel",
        "papel":"empate",
        "tesoura":"tesoura"
    },
    "tesoura":{
        "pedra":"pedra",
        "papel":"tesoura",
        "tesoura":"empate"
    }
    };
    return tabelaDeJogadas[jogada1][jogada2];
}

function jogadorJokenpo(jogador1, jogada1, jogador2, jogada2)
{

    if(jogada2 == "tesoura")
        return "Pedro";
    if(jogada2 == "pedra")
        return "empate;"
    return "João";
}
